### PR TITLE
sharing with user with uppercase username

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

acceptance test(webui) for sharing file with user with username containing uppercase letters
## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Review by skshetry — COMMENTED** [review]
:+1:

**Comment by phil-davis** [comment]
This provides a test scenario for issue #35382

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 1]
```diff
@@ -280,3 +280,15 @@ Feature: Sharing files and folders with internal users
     And user "user1" has shared file "lorem.txt" with user "user2"
     When the user opens the share dialog for file "lorem.txt"
     Then the user should not be able to send the share notification by email using the webUI
+
+  @issue-35382
+  Scenario: user shares a file with another user with uppercase username
+    Given these users have been created:
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    And these users have been created without skeleton files:
      | username  |
      | SomeUser |
```

Please rebase and user this step instead.

**Comment by phil-davis** [comment]
It got drone CI "cannot reach github.com" errors.
I restarted drone just now.

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: @paurakhsharma why change `Given`  to `And` ?

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 2]
```diff
@@ -280,3 +280,15 @@ Feature: Sharing files and folders with internal users
...
+      | username |
+      | SomeUser |
+    And user "user1" has logged in using the webUI
+    When the user shares file "lorem.txt" with user "SomeUser" using the webUI
+    And the user re-logs in as "SomeUser" using the webUI
+    And the user browses to the shared-with-you page
+    Then file "lorem.txt" should not be listed on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    Then file "lorem (2).txt" should not be listed on the webUI
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 3]
```diff
@@ -280,3 +280,15 @@ Feature: Sharing files and folders with internal users
...
+#    Then file "lorem.txt" should be listed on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
#    Then file "lorem (2).txt" should be listed on the webUI
```

**Review by phil-davis — CHANGES_REQUESTED** [review]
When it works properly, we expect to see `lorem (2).txt` because the user already has `lorem.txt` from their skeleton.

I ran this scenario in `release-10.2.0` branch and the problem is not in that branch, so the regression has happened in some PR since `10.2.0`  was released.

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 1 above)
Comment (paurakhsharma) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: Oh my fault, that should be given

**Review by paurakhsharma — COMMENTED** [review]
Looks good to me :+1:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 2 above)
Comment (phil-davis) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: @HariBhandari07 IMO when this works we expect to see `lorem (2).txt` on the webUI.
So we want to check `lorem (2).txt`

Comment (individual-it) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: `SomeUser` has been created without a skeleton folder

Comment (phil-davis) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: OK, now I understand

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@HariBhandari07 please backport to `stable10`

**Comment by HariBhandari07** [comment]
Backport on #35409

**Pull request merged**